### PR TITLE
FTUE: Smart persisted expanded state

### LINKED ISSUE #5971: Initial Help Companion State Open When Unread Tips Present
## Description 

After further discussion between @aaecheverria and @samitron7  - adding conditional open state as initial state of helper companion (the help center with quick tips in it) to be _open_ or _expanded_ when there are unread tips present. 

## Acceptance Criteria 
- If a user has unread tips in their helper companion the box should have the initial state be be expanded/open
Screen Shot 2021-01-19 at 8 49 58 AM

- If a user has read all tips in their helper companion the box should have the initial state be collapsed/closed  Screen Shot 2021-01-15 at 4 12 27 PM
- If a user has unread tips but has dismissed the helper companion, that toggle should be stored in local storage and respected on future visits to the editor (see thread for additional context) 
- Work should include updated unit tests

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Requested enhancement for FTUE opening/closing behavior to have smart persistence.

## Summary
Adds smart opening behavior requested in ticket & unit tests.

## Relevant Technical Choices
Realized initial state could follow our effects pattern, it's just a single composed effect that runs on initialization tho where `previous` is the persisted state and `next` is the initial state. 

Allows us to only have to test the pure fn effects since every other mechanism already has testing from this PR: https://github.com/google/web-stories-wp/pull/6401

## To-do
NA

## User-facing changes
FTUE now has smart persistence for opening/closing. so it takes into account your last state in the last browser session as well as some other logic to compute if it should initialize to open or not.

## Testing Instructions
-> Go to editor and see ftue toggle.
-> With all your tips read, make sure the companion is collapsed
-> copy and paste this into your console (`cmnd + option + i` to open) and press `enter` to reset your read tips on the endpoint: 
```
await wp.apiFetch({ path: '/web-stories/v1/users/me', method: 'POST', data: { meta: { web_stories_onboarding: {} }}})
```
-> refresh your browser. FTUE should now auto animate open when it receives a response from the Backend telling it there are new unread tips
-> read 1 or two tips, but not all.
-> leave ftue companion open and refresh page. ftue companion should load open on mount
-> Close ftue companion and refresh page. ftue companion should load closed on mount
-> open the ftue companion and read through all the tips
-> with the companion open, refresh the browser. ftue companion should load closed/collapsed on mount

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Follow testing instructions above

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #5971

### CODE REVIEW

**Review by BrittanyIRL — COMMENTED** [review]
I love how much this tied in to the existing structure! 

Works great, except for if I leave the page go somewhere else in my WordPress dashboard and come back.

So like, if i have FTUE open and 5 unread tips and  click 'appearance' and go look at some themes and come back to create a new story, i'd expect FTUE to be open when i get to the page, but it's collapsed.

File: assets/src/edit-story/components/helpCenter/useHelpCenter/test/effects.js [hunk 1]
```diff
@@ -16,15 +16,66 @@
...
-import { initial, deriveState } from '../';
+import { initialState, deriveState } from '../';
 
 const mockState = (overrides) => ({
-  ...initial.state,
+  ...initialState,
   ...overrides,
 });
 
+describe('deriveInitialOpen', () => {
```
Comment (embarks) [inline assets/src/edit-story/components/helpCenter/useHelpCenter/test/effects.js:89]: tests are so nice

**Comment by spacedmonkey** [comment]
Everytime I open the editor, I am seeing a POST request to the current user API. Even if I haven't touched the FTUE. Why is this needed? Is this an expected bahaviour?

Screenshot 2021-02-19 at 18 50 00

This is also happening in main, should I create another issue for this or do you want to fix this here?

File: package-lock.json [hunk 2]
```diff
@@ -2475,7 +2475,6 @@
         "jest-resolve": "^26.6.2",
         "jest-util": "^26.6.2",
         "jest-worker": "^26.6.2",
-        "node-notifier": "^8.0.0",
```
Comment (spacedmonkey) [inline package-lock.json]: Are these changes needed?

File: assets/src/edit-story/utils/localStore.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ export const LOCAL_STORAGE_PREFIX = {
   PANEL: 'web_stories_ui_panel_settings',
   TEXT_SET_SETTINGS: 'web_stores_text_set_settings',
   TERMS_MEDIA3P: 'web_stories_media3p_terms_agreement',
+  FTUE: 'web_stories_ftue',
```
Comment (swissspidy) [inline assets/src/edit-story/utils/localStore.js]:
```suggestion
  HELP_CENTER: 'web_stories_help_center',
```

Seems more suitable

File: package-lock.json (hunk 2 above)
Comment (maxyinger) [reply package-lock.json]: nope. not intentional. I'll get rid of them

**Comment by maxyinger** [comment]
> Everytime I open the editor, I am seeing a POST request to the current user API. Even if I haven't touched the FTUE. Why is this needed? Is this an expected bahaviour?
> 
> Screenshot 2021-02-19 at 18 50 00

I'll clean up in here. just an effect that needs some extra conditions

**Comment by maxyinger** [comment]
> I love how much this tied in to the existing structure!
> 
> Works great, except for if I leave the page go somewhere else in my WordPress dashboard and come back.
> 
> So like, if i have FTUE open and 5 unread tips and click 'appearance' and go look at some themes and come back to create a new story, i'd expect FTUE to be open when i get to the page, but it's collapsed.

Okay, so looking at this, it's sort of a side effect of closing on blur. It's not an issue if you navigate with just the nav bar, or history buttons, but if you click a link within the page, it causes a blur and closes, then persists that closed state.

I don't think we should add some complex logic to close on some blur cases and not close on others, I think we should just either choose to close on blur or not. and understand this is behavior is part of closing on blur

**Comment by BrittanyIRL** [comment]
> I don't think we should add some complex logic to close on some blur cases and not close on others, I think we should just either choose to close on blur or not. and understand this is behavior is part of closing on blur

I think this makes sense. I'm fine with whichever, it just feels as though it should have consistency. Definitely not something that needs to get handled in this PR. Would you mind following up with Tad and Sam to figure out what way this should be handled or if the inconsistency is fine with them?

**Comment by maxyinger** [comment]
@BrittanyIRL alright Tad said it's cool to just remove the close on focus out behavior, so should no longer run into the issue you mentioned: https://xwp.slack.com/archives/C01ESBB8ZN1/p1614011120011000?thread_ts=1614010050.007200&cid=C01ESBB8ZN1

File: assets/src/edit-story/components/helpCenter/useHelpCenter/index.js [hunk 4]
```diff
@@ -175,15 +190,48 @@ export function useHelpCenter() {
     [store.actions]
   );
 
+  // Once app is hydrated, start persisting unreadTips
+  const { isHydrated, unreadTipsCount } = store.state;
+  useEffect(() => {
+    if (isHydrated) {
+      const local = localStore.getItemByKey(LOCAL_STORAGE_PREFIX.FTUE);
```
Comment (spacedmonkey) [inline assets/src/edit-story/components/helpCenter/useHelpCenter/index.js]: You need to fix  all the references to `LOCAL_STORAGE_PREFIX.FTUE` to `LOCAL_STORAGE_PREFIX.HELP_CENTER`

**Comment by spacedmonkey** [comment]
The lint and test errors seem valid, can you look at those please.

File: assets/src/edit-story/components/helpCenter/useHelpCenter/index.js (hunk 4 above)
Comment (maxyinger) [reply assets/src/edit-story/components/helpCenter/useHelpCenter/index.js]: Ope, yeaup

Comment (maxyinger) [reply assets/src/edit-story/components/helpCenter/useHelpCenter/index.js]: Thanks for catching

**Review by BrittanyIRL — APPROVED** [review]
Thanks for sorting out the focusOut state! 

LGTM.

**Review by embarks — APPROVED** [review]
Great job, very clear what's going on and how it's supposed to behave. Testing went well. :)

**Comment by csossi** [comment]
Verified in QA
